specs/sequencer-app.md

```diff
@@ -41,143 +42,306 @@ idea of "messages", where every transaction can contain multiple messages. The
-
-The first, part of the `accounts` component, is a value-transfer action:
-
-```rust
-pub struct Transfer {
-    to: Address,
-    amount: Balance,
+pub(crate) struct Actions {
+    group: Group, // actions are grouped by "bundleability" and authority level
+    inner: Vec<Action>,
 }
 ```
 
-This action transfers the specified amount funds from the sender to the
-recipient.
+`Nonce` is an incrementing value that represents how many transactions have been
+previously submitted by the currently submitting account; thus it starts at 0,
+and must strictly increase by 1 for each following transaction.
 
-The second, part of the `sequence` component, accepts generic data with a
-specified chain ID:
+The structure of a (signed) transaction is as follows:
 
 ```rust
-pub struct Action {
-    pub(crate) chain_id: Vec<u8>,
-    pub(crate) data: Vec<u8>,
+pub struct Transaction {
+    /// transaction signature
+    signature: ed25519_consensus::Signature,
+    /// the verification key of the signer
+    verification_key: ed25519_consensus::VerificationKey,
+    /// the transaction body which is signed
+    body: TransactionBody,
+    /// a bytes representation of the transaction body
+    body_bytes: Bytes,
 }
 ```
 
-Note that this action does not have any effect on the state of the sequencer; it
-is simply ordered by the sequencer and placed into a block.
+The address corresponding to the signer is derived from the
+`ed25519_consensus::VerificationKey` (i.e. the public key).
 
-## ABCI block lifecycle
+## Actions
+
+The following is an exhaustive list of all user-accessible actions available
+to be submitted as of Sequencer v1.0.0-rc.1.
+
+### Core Protocol Actions
+
+* `Transfer`: represents a value transfer action between two accounts. It consists
+of the following fields:
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | to        | `Address`| The account to transfer to. The "from" address is assumed|
+  ||| to be the signer of the transaction. |
+  | amount    | `uint128`| The amount to transfer. |
+  | asset     | `string` | The asset to transfer. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+
+* `RollupDataSubmission`: a transaction ordered by the sequencer, whose destination
+is another chain. It consists of the following:
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | rollup_id | `RollupId`| ID of the destination chain. |
+  | data      | `bytes`  | The opaque transaction data. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+
+### Bridge Actions
+
+These actions deal with transfering funds to/from a bridge account to be used on
+a rollup.
+
+* `InitBridgeAccount`: initializes a bridge account for the given rollup on the
+sequencer chain. The signer of the transaction is the owner of the bridge account,
+and the only actor authorized to transfer out of the account. It is made up of
```

Comment:
>and the only actor authorized to transfer out of the account
This is the `withdrawer_address`. By default it's set to the signer of the bridge init action, but can be set to a different address (either by providing a value here, or using a `BridgeSudoChange` action afterwards).

Reply:
Updated the description in fd47dc7. Lmk if that works for you!